server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpRequestBody.scala

```diff
@@ -66,4 +76,55 @@ class SttpRequestBody[F[_]](implicit ME: MonadError[F]) extends RequestBody[F, A
     transfer()
     os.toByteArray
   }
+
+  private def extractMultipartParts(parts: Seq[Part[client3.RequestBody[_]]], bodyType: RawBodyType.MultipartBody): List[Part[Any]] = {
+    parts.flatMap { part =>
+      bodyType.partType(part.name).flatMap { partType =>
+        extractPartBody(part, partType).map { body =>
+          Part(
+            name = part.name,
+            body = body,
+            contentType = part.contentType.flatMap(ct => MediaType.parse(ct).toOption),
+            fileName = part.fileName
+          )
+        }
+      }
+    }.toList
+  }
+
+  private def extractPartBody[B](part: Part[client3.RequestBody[_]], bodyType: RawBodyType[B]): Option[Any] = {
+    part.body match {
+      case ByteArrayBody(b, _) =>
+        bodyType match {
+          case RawBodyType.StringBody(charset)  => Some(b)
+          case RawBodyType.ByteArrayBody        => Some(b)
+          case RawBodyType.ByteBufferBody       => Some(ByteBuffer.wrap(b))
+          case RawBodyType.InputStreamBody      => Some(new ByteArrayInputStream(b))
+          case RawBodyType.InputStreamRangeBody => Some(InputStreamRange(() => new ByteArrayInputStream(b)))
+          case RawBodyType.FileBody             => throw new IllegalArgumentException("ByteArray body provided while endpoint accepts FileBody")
+          case _: RawBodyType.MultipartBody     => None
+        }
+      case FileBody(f, _) =>
+        bodyType match {
+          case RawBodyType.FileBody        => Some(FileRange(f.toFile))
+          case RawBodyType.ByteArrayBody   => Some(Files.readAllBytes(f.toPath))
+          case RawBodyType.ByteBufferBody  => Some(ByteBuffer.wrap(Files.readAllBytes(f.toPath)))
+          case RawBodyType.InputStreamBody => Some(new FileInputStream(f.toFile))
+          case _                           => None
+        }
+      case StringBody(s, charset, _) =>
+        bodyType match {
+          case RawBodyType.StringBody(_)  => Some(s)
+          case RawBodyType.ByteArrayBody  => Some(s.getBytes(charset))
+          case RawBodyType.ByteBufferBody => Some(ByteBuffer.wrap(s.getBytes(charset)))
+          case _                          => None
+        }
+      case InputStreamBody(is, _) =>
+        bodyType match {
+          case RawBodyType.InputStreamBody => Some(is)
+          case _                           => None
```

Comment:
there's still a lot of `None`s that are returned. Plus, instead of throwing, we should probably use `ME.error`, just as in `toRaw`. I don't think returning an `Option` is requried here, as if the body doesn't match, we should return an error